middleware/src/main/java/com/nytimes/android/external/store3/middleware/GsonReaderParser.kt

```diff
@@ -0,0 +1,28 @@
+package com.nytimes.android.external.store3.middleware
+
+import com.google.gson.Gson
+import com.nytimes.android.external.store3.base.Parser
+import com.nytimes.android.external.store3.util.ParserException
+
+import java.io.Reader
+import java.lang.reflect.Type
+
+import javax.inject.Inject
+
+import io.reactivex.annotations.NonNull
+
+import com.nytimes.android.external.cache3.Preconditions.checkNotNull
+
+class GsonReaderParser<Parsed> @Inject
+constructor(private val gson: Gson, private val type: Type) : Parser<Reader, Parsed> {
+
+    init {
+        checkNotNull(gson, "Gson can't be null")
```

Comment:
This will not work due to the fact that Kotlin will fail with a KotlinNPE in case you pass a null value for `gson` or `type`. In that case, the checks in `init` are obsolete :)

Reply:
done!